CONTRIBUTING.md

```diff
@@ -1,256 +1,263 @@
+this repository.
 
+```shell
+git clone https://github.com/jupyterhub/zero-to-jupyterhub-k8s
+```
 
+# Setting up for documentation development
 
-## Local development
+See [doc/README.md](doc/README.md).
 
-### 1: Preparations
+# Setting up for Helm chart development
 
-Before anything else, install [git](https://www.git-scm.com/), clone the
-repository, and enter the repository directory.
 
-```
-git clone https://github.com/jupyterhub/zero-to-jupyterhub-k8s
-cd zero-to-jupyterhub-k8s
-```
 
-For local development, you will additional tools and we present you with two
-options. Either you, a) start and work from a Virtual Machine that is
-automatically prepared for development, or b) install the tools yourself and
-work without a Virtual Machine.
+## 1: Prerequisites
 
-#### a) Use a prepared Virtual Machine (VM)
+This needs to be installed:
 
-1. Install VirtualBox by [downloading and running an
-   installer](https://www.virtualbox.org/wiki/Downloads).
+- [docker](https://docs.docker.com/get-docker/)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) (also [setup tab completion](https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion))
+- [helm](https://helm.sh/docs/intro/install/) (also [setup tab completion](https://helm.sh/docs/helm/helm_completion/))
+- Python 3.7+ ([Anaconda.com](https://www.anaconda.com/distribution/),
+  [Python.org](https://www.python.org/downloads/)) and dependencies:
+  ```shell
+  pip install -r dev-requirements.txt
+  pip install -r doc/doc-requirements.txt
+  ```
 
-1. Install Vagrant by [downloading and running an
-   installer](https://www.vagrantup.com/downloads.html).
+You can check if you have it all like this:
 
-1. Start and automatically setup a VM.
+```shell
+docker --version
+kubectl version --client
+helm version
+pytest --version
+chartpress --version
+```
 
-   Use the `vagrant up` command to start the VM for the first time, or `vagrant
-   resume` to resume a suspended VM.
+## 2: Setup a Kubernetes cluster
 
-   ```shell
-   vagrant up
-   # vagrant resume
-   ```
+We need a Kubernetes cluster to work against.
 
-1. Enter the VM and the `~/zero-to-jupyterhub-k8s` folder.
+### Linux only: Kubernetes setup with k3s
+With [k3s](https://github.com/rancher/k3s) we can _quickly_ create a Kubernetes
+cluster, and we _don't have to transfer docker images_ built on our computer to
+make them available in the Kubernetes cluster.
 
-   The `~/zero-to-jupyterhub-k8s` folder in the VM will be the exact same folder
-   on your actual machine. Change either and you influence both.
+__Install__
 
-   ```shell
-   vagrant ssh
-   cd zero-to-jupyterhub-k8s
-   ```
+```shell
+# Installs a ~50 MB k3s binary, setups and starts a systemctl service called
+# k3s which is also enabled to run on startup, provides a k3s-uninstall.sh
+# script, disables not needed functionality. You will be asked for sudo rights
```

Comment:
I'd be fine with that potential issue, since k3s is in the CI system + I'd use it for local dev so I feel confident we would catch issues quite quick.